### DISCUSSION TITLE
Support deno

### DISCUSSION THREAD

**Discussion by deathemperor** [serverless/serverless#9785] [General]
Any thought on this?

Feature request on serverless-offline repo https://github.com/dherault/serverless-offline/issues/1258

**Comment by pgrzesik** [comment]
Hello @deathemperor - could you please share reasoning behind potentially supporting `deno`? Also - could you clarify what do you mean by that? Are you talking about supporting `deno` as runtime for functions or the Framework itself being compatible with deno?

**Comment by deathemperor** [comment]
Hello, my bad I didn't make it clear on my first issue. But my questions are:

1. Could/would serverless support deno runtime? I know there's repo like this one https://github.com/hayd/deno-lambda but is there any chance for official support? So it's a bit of both: `deno` as runtime for functions and the frameworks itself being compatible with `deno`
2. Plugins: this part is big. The nodejs serverless ecosystem is huge and very mature. There are certain plugins stopping me from using deno for serverless: serverless-offline for example.

I'm not sure if other plugins are compatible, guess it depends on what that plugin does?

Bottom line: I really want to start using `deno` for serverless. What I've described above are what I think limits that need to be removed before that can be reality.

**Comment by pgrzesik** [comment]
Thanks for clarification @deathemperor :bow: 

Related to 1. It definitely could support deno runtime, but that would have to be provided either as a custom runtime or you could use container support to deploy deno app. As for runtime, it's definitely not a part of core Framework 

Related to 2. Here I'm assuming that you're only talking about Deno as a Lambda runtime - I'm not sure what plugins would be supported, I would assume a lot of them would be, as long as they're not runtime-specific such as `serverless-python-requirements`. Serverless-offline can also be problematic, but I think it might be possible to also provide some kind of Deno runtime for it assuming it's available

If you had in mind the Framework as a tool running on Deno instead of Node then I don't think it's really an option at this point.

**Comment by deathemperor** [comment]
@pgrzesik deployment works pretty OK for now with hayd/deno-lambda. The biggest concern is serverless-offline, without it development couldn't work. I'm patiently waiting for a solution still, as Deno is still very early.

thanks for addressing the questions, I appreciate it.

**Comment by pgrzesik** [comment]
Would you mind submitting an issue to `serverless-offline` repository related to that? Also, I'm going to move this issue to discussions as I believe it's better suited there - we're also looking forward to hearing from other people that are interested in Deno support :+1:

**Reply by deathemperor** [reply]
I've add a feature request on serverless-offline https://github.com/dherault/serverless-offline/issues/1258

**Comment by Alon2303** [comment]
Hi, any development on this?

**Reply by pgrzesik** [reply]
Hello @Alon2303 - could you clarify what type of support would you like to see?

**Reply by Alon2303** [reply]
Hi @pgrzesik I read the discussion and wondered maybe there were any development in the past few months regarding official support for deno run time? maybe in some future release?
thx

**Reply by pgrzesik** [reply]
The deno runtime is supported via custom runtimes just like any other runtime that isn't officialy supported by AWS - it can also be used via Lambda container images. Example of deno runtime use - https://github.com/hayd/deno-lambda/tree/master/example-serverless

Is there anything else that you have in mind? Also, just a note, the interest in Deno runtime support is quite small, so I wouldn't expect AWS to support it natively anytime soon.
